pyproject.toml

```diff
@@ -9,6 +9,7 @@ packages = [
 
 [tool.poetry.dependencies]
 python = ">=3.9,<3.13"
+memory_profiler = "^0.58.0"
```

Comment:
@smngvlkz This shouldn't be a core dependency—it's only used when performing benchmarking.